frame/nomination-pools/test-staking/src/mock.rs

```diff
@@ -0,0 +1,261 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
```

Comment:
i wonder if you could cut down on duplicated stuff by importing the mock from the main repo

Reply:
anything that's `#cfg(test)` cannot be exported unfortunately. We could abuse `std` or a new feature for this, but I've tried it before and the result was not pleasant.